tests/results.cpp

```diff
@@ -2027,3 +2027,212 @@ TEST_CASE("distinct") {
         REQUIRE(further_filtered.get(0).get_int(2) == 9);
     }
 }
+
+
+TEST_CASE("aggregate") {
+#define SECTIONS_RESULT_BUILT_FROM_TABLE_QUERY_TABLE_VIEW() \
+    SECTION("results built from table") { \
+        results = Results(r, *table); \
+    } \
+    SECTION("results built from query") { \
+        results = Results(r, table->where()); \
+    } \
+    SECTION("results built from tableview") { \
+        results = Results(r, table->where().find_all()); \
+    }
+
+    const int column_count = 4;
+    InMemoryTestFile config;
+    config.cache = false;
+    config.automatic_change_notifications = false;
+
+
+    auto r = Realm::get_shared_realm(config);
+    r->update_schema({
+        {"object", {
+            {"int", PropertyType::Int, "", "", false, false, true},
+            {"float", PropertyType::Float,  "", "", false, false, true},
+            {"double", PropertyType::Double, "", "", false, false, true},
+            {"date", PropertyType::Date, "", "", false, false, true},
+        }},
+    });
+
+    auto table = r->read_group().get_table("class_object");
+
+    SECTION("one row with null values") {
+        r->begin_transaction();
+        table->add_empty_row(3);
+        for (int i = 0; i < column_count; ++i) {
+            table->set_null(i, 0);
+        }
+
+        table->set_int(0, 1, 0);
+        table->set_float(1, 1, 0.f);
+        table->set_double(2, 1, 0.0);
+        table->set_timestamp(3, 1, Timestamp(0, 0));
+
+        table->set_int(0, 2, 2);
+        table->set_float(1, 2, 2.f);
+        table->set_double(2, 2, 2.0);
+        table->set_timestamp(3, 2, Timestamp(2, 0));
+        // table:
+        //  null, null, null,  null,
+        //  0,    0,    0,    (0, 0)
+        //  2,    2,    2,    (2, 0)
+        r->commit_transaction();
+
+        SECTION("max") {
+            Results results;
+
+            SECTIONS_RESULT_BUILT_FROM_TABLE_QUERY_TABLE_VIEW()
+
+            REQUIRE(results.max(0)->get_int() == 2);
+            REQUIRE(results.max(1)->get_float() == 2.f);
+            REQUIRE(results.max(2)->get_double() == 2.0);
+            REQUIRE(results.max(3)->get_timestamp() == Timestamp(2, 0));
```

Comment:
Test fails because of core's `Table::maximum_timestamp().is_null() == true` which seems to be a bug. checking now.